src/Module/RenderCartBookingInfoHandler.php

```diff
@@ -0,0 +1,207 @@
+     * @param object                                        $exprBuilder      The expression builder.
+     * @param array|stdClass|ArrayAccess|ContainerInterface $cartItemConfig   The cart item data config.
+     */
+    public function __construct(
+        SelectCapableInterface $bookingsSelectRm,
+        $exprBuilder,
+        $cartItemConfig
+    ) {
+        $this->bookingsSelectRm = $bookingsSelectRm;
+        $this->exprBuilder      = $exprBuilder;
+        $this->cartItemConfig   = $cartItemConfig;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $item = $event->getParam(0);
+        $item = $this->_normalizeContainer($item);
+
+        $dataKey       = $this->_containerGetPath($this->cartItemConfig, ['data', 'key']);
+        $eddBkKey      = $this->_containerGetPath($this->cartItemConfig, ['data', 'eddbk_key']);
+        $bookingIdKey  = $this->_containerGetPath($this->cartItemConfig, ['data', 'booking_id_key']);
+        $bookingIdPath = [$dataKey, $eddBkKey, $bookingIdKey];
+
+        try {
+            $bookingId = $this->_containerGetPath($item, $bookingIdPath);
+        } catch (NotFoundExceptionInterface $exception) {
+            // Cart item does not have a booking ID, and thus does not correspond to a booking.
+            return;
+        }
+
+        // Alias expression builder
+        $b = $this->exprBuilder;
+
+        // Fetch the corresponding booking from storage
+        $condition = $b->eq($b->var('id'), $b->lit($bookingId));
+        $bookings  = $this->bookingsSelectRm->select($condition);
+        // Stop if no booking was found, or if multiple bookings matched the ID (for some reason?)
+        if ($this->_countIterable($bookings) !== 1) {
+            return;
+        }
+
+        // Get the booking
+        $booking = reset($bookings);
+
+        echo $this->_renderBookingInfo($booking);
+    }
+
+    /**
+     * Renders the information for a booking.
+     *
+     * @since [*next-version*]
+     *
+     * @param BookingInterface $booking The booking.
+     *
+     * @return string|Stringable The render result.
+     */
+    protected function _renderBookingInfo(BookingInterface $booking)
+    {
+        $format = $this->_containerGet($this->cartItemConfig, 'booking_datetime_format');
+
+        $startTs  = $booking->getStart();
+        $startDt  = date(DATE_ATOM, $startTs);
+        $startStr = date($format, $startTs);
+
+        $endTs  = $booking->getEnd();
+        $endDt  = date(DATE_ATOM, $endTs);
+        $endStr = date($format, $endTs);
+
+        $startLine = sprintf(
+            '<b>%1$s</b> <time datetime="%2$s">%3$s</time>',
```

Comment:
What if, instead of creating HTML strings, you use an injected placeholder template? Souds much more SoCy, and configurable.

Reply:
Where could I store the template?

Reply:
Service goes to DI container, configuration goes to config.

Reply:
I anticipate that one day (hopefully soon), we will swap out the implementation to a PHP template, and nothing in this handler will need to change.

Reply:
I was referring to the template contents: the actual template string.

Reply:
I think, the best idea is to read it from file.